src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -282,17 +309,22 @@ public void postStop() throws Exception {
     private long _postRequests = 0;
     private boolean _waiting = false;
     private final int _maximumConcurrency;
-    private final EvictingQueue<Request> _pendingRequests;
+    private final EvictingQueue<ImmutableMap<Request, Long>> _pendingRequests;
```

Comment:
Why is this `ImmutableMap<Request, Long>`? It looks like you're trying to package up a `Request` with the time at which it was enqueued?

A map isn't a great choice here - you intend only to ever populate one key in it, which is pretty confusing to readers, and it's an invariant that the compiler can't help you enforce.

I'm not sure what the _best_ Java idiom is here. Some options:
* A `class RequestQueueEntry` that contains the original Request and its start timestamp as members.
* One of the "pair" options outlined here: https://stackoverflow.com/questions/521171/a-java-collection-of-value-pairs-tuples

Maybe ask around the team to see if someone more familiar with Java libraries has a better answer.